src/engine/helper/ec_local.h

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright 2001-2019 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright (c) 2002, Oracle and/or its affiliates. All rights reserved
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <stdlib.h>
+#include <openssl/obj_mac.h>
+#include <openssl/ec.h>
+#include <openssl/bn.h>
+
+/* Note: Although these structures are internal to OpenSSL, they are defined
+ *       here, because in order to override ecc point operations with pka
+ *       API's the fields inside the structure have to be changed directly as
+ *       OpenSSL doesn't provide API's to do so.
+ *
+ *       Care needs to be taken to preserve the order and members of the structure
+ *       same as in the respective versions of OpenSSL.
+ *
+ *       Structures are mirrored from crypto/ec/ec_local.h inside openssl github repo.
+ *       Prior to openssl 1.1.1 version, this file is named as ec_lcl.h
```

Comment:
Is it an exact copy of the file ec_lcl.h, or only used structures are copied?

Reply:
Only the used structures are copied. There are many other definitions which require inclusion of other internal headers. These other definitions are not useful to us. Hence, only used structures are copied.